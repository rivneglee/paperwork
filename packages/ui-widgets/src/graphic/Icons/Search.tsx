/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' } : { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1070 1024">
    <path
      d="M675.9168 242.816C556.3008 123.2 362.4448 123.2 242.8288 242.816 123.2128 362.432 123.2128 556.288 242.8288 675.904 362.4448 795.52 556.3008 795.52 675.9168 675.904 795.5328 556.288 795.5328 362.432 675.9168 242.816L675.9168 242.816ZM1000.7168 892.48C1030.6048 922.368 1030.6048 970.816 1000.7168 1000.704 970.8288 1030.656 922.3808 1030.656 892.4288 1000.704L730.0608 838.336C728.1408 836.416 727.4368 833.92 725.7728 831.872 546.3168 960.448 295.7568 945.408 134.5408 784.192-44.8512 604.8-44.8512 313.92 134.5408 134.528 313.9328-44.864 604.8128-44.864 784.2048 134.528 945.4208 295.744 960.4608 546.304 831.8848 725.76 833.9328 727.424 836.4288 728.128 838.3488 730.048L1000.7168 892.48Z"
    ></path>
  </svg>
);
