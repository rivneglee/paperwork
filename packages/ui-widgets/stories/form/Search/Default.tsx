import React from 'react';

import { Search, Card } from '../../../src';

export default () => (
  <div style={{ background: '#ebeef1', padding: '2.4rem' }}>
    <Card header={<h3>Input types</h3>}>
      <Search placeholder="Search..."/>
    </Card>
  </div>
);
