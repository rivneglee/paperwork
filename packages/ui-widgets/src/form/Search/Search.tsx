import React, { FunctionComponent, KeyboardEvent, SyntheticEvent } from 'react';
import classNames from 'classnames';

import { Input } from '../Input';
import Icons from '../../graphic/Icons';
import { Button } from '../Button';

interface Props {
  value?: string | number;
  onChange?: (e: SyntheticEvent) => void;
  onApply?: () => void;
  placeholder?: string;
  className?: string;
  applyButtonText?: string;
}

const Search: FunctionComponent<Props> = ({
  value,
  onChange,
  onApply,
  placeholder,
  className,
  applyButtonText = 'Apply',
}: Props) => {
  const onKeyDown = (e: KeyboardEvent) => {
    if (e.keyCode === 13 && onApply) {
      onApply();
    }
  };
  return (
    <Input
      className={
        classNames(
          'pw-search',
          className,
        )
      }
      value={value}
      onChange={onChange}
      onKeyDown={onKeyDown}
      placeholder={placeholder}
      left={<Icons.Search />}
      right={<Button onClick={onApply} className="pw-search__apply" type="link">{applyButtonText}</Button>}
    />
  );
};

export default Search;
