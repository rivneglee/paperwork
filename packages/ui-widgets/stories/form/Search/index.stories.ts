import { storiesOf } from '@storybook/react';

import Default from './Default';

storiesOf('Search', module)
  .add('default', Default);
