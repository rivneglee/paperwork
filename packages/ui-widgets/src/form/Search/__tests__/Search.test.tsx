import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Search from '../Search';

describe('Search', () => {
  it('should render properly', () => {
    const wrapper: ShallowWrapper = shallow(<Search placeholder="abc" value="value" className="extra-class" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
