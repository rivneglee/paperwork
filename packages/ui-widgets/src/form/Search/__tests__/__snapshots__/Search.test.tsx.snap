// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render properly 1`] = `
<Input
  className="pw-search extra-class"
  left={<Unknown />}
  onKeyDown={[Function]}
  placeholder="abc"
  right={
    <Button
      className="pw-search__apply"
      onClick={undefined}
      type="link"
    >
      Apply
    </Button>
  }
  value="value"
/>
`;
